### PR TITLE
Mesh .updateMorphTargets(): Added BufferGeometry support

### PR DESCRIPTION
**Pull request by Mugen87**
see #11277

### CODE REVIEW

File: src/objects/Mesh.js [hunk 1]
```diff
@@ -56,17 +56,53 @@ Mesh.prototype = Object.assign( Object.create( Object3D.prototype ), {
...
+				var morphAttribute = morphAttributes[ keys[ 0 ] ];
+
+				if ( morphAttribute !== undefined ) {
+
+					this.morphTargetInfluences = [];
+					this.morphTargetDictionary = {};
+
+					for ( m = 0, ml = morphAttribute.length; m < ml; m ++ ) {
+
+						name = morphAttribute[ m ].name || m;
```
Comment (looeee) [inline src/objects/Mesh.js]: Would it be a good idea to make sure the name is always a string? Or do you think it doesn't matter?

Comment (Mugen87) [inline src/objects/Mesh.js]: Yeah, maybe it's cleaner if we always ensure strings. I'll change this...

**Comment by looeee** [comment]
Might need to make some updated to `PropertyBindings.js` too, as it has a special case for `morphTargetInfluences` [here](https://github.com/mrdoob/three.js/blob/dev/src/animation/PropertyBinding.js#L607)...

**Comment by Mugen87** [comment]
I need to debug this in order to understand the code. We should currently run into this log [here](https://github.com/mrdoob/three.js/blob/dev/src/animation/PropertyBinding.js#L621) because `morphTargets` is `undefined`. Maybe @tschw can help here.

**Comment by looeee** [comment]
By the way I had a search  through code in /src and /examples/js and couldn't find any other places where this will cause an issue. Although I may have missed something of course!

**Comment by Mugen87** [comment]
@looeee It is important, that the name of the morph attribute matches the name of the KeyframeTrack. Watch at the nomenclature `propertyName[attributeName]`.

```javascript
const morphAttribute = new THREE.BufferAttribute( morphPositions, 3 );
morphAttribute.name = 'animation001';
geometry.morphAttributes.position.push( morphAttribute );
```

```javascript
const track = new THREE.NumberKeyframeTrack( '.morphTargetInfluences[animation001]', [ 0.0, 5 ], [ 0.0, 1.0 ], THREE.InterpolateSmooth );
```
`morphTargetInfluences` can be affected by not just a single track of morph target data. A more complex animation consists of many morph targets. The suffix of a name is normally consecutively numbered to represent the order. This information is used by `PropertyBinding`. The last PR should added support for `BufferGeometry` based morphTargets in `PropertyBinding`. But this needs a code review by @tschw

**Comment by looeee** [comment]
So, the `name` property in `KeyframeTrack` will be slightly different depending on whether it's targeting a `morphTarget` or a `morphAttribute`?

**Comment by Mugen87** [comment]
No they are identical. Normally, clips are created not manually but like this:

```javascript
// Create clips with THREE.AnimationClip
// We assume geometry.morphAttributes was created by a loader

const clips = THREE.AnimationClip.CreateClipsFromMorphTargetSequences( geometry.morphAttributes.position, 10 );
const clip = THREE.AnimationClip.findByName( clips, 'animation' );

const mixer = new THREE.AnimationMixer( mesh );
const animationAction = mixer.clipAction( clip );
```

**Comment by Mugen87** [comment]
Check out the implementation of `AnimationClip .CreateClipsFromMorphTargetSequences` to see, how the naming convention works. There are some helpful comments in the code.

**Comment by mrdoob** [comment]
Should we also remove the hack in `MMDLoader`? https://github.com/mrdoob/three.js/issues/11277#issuecomment-299359144

/cc @takahirox

**Comment by takahirox** [comment]
I haven't follow the entire discussion yet but I suppose
we can remove if facial morphing animation of `webgl_loader_mmd*` examples work fine without the hack.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**